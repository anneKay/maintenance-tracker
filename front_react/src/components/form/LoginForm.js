import React from 'react';
import { Form, Button } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import validator from 'validator';
import InlineError from '../messages/InlineError';
import propTypes from 'prop-types';
class UserForm extends React.Component {
    state = {
      loading: false,
			data:{
				email: '',
				password: ''
			},
			errors:{}
		};
		
		onChange = (event) => 
		this.setState({ 
			data: { email:this.state.data.email, password:this.state.data.password, [event.target.name] : event.target.value }
		})

		validate = (data) => {
			const errors = {};
			if (!data.password) errors.password = 'password cannot be blank';
			if (!validator.isEmail(data.email)) errors.email = 'invalid email';
			return errors;
		}

		onSubmit = () => {
			const errors = this.validate(this.state.data);
			const { submit, history} = this.props;
			this.setState({
				errors,
			})
			if(Object.keys(errors).length === 0 ){
				submit(this.state.data, history);
			}
		}

 
    render () {
			const {data, errors} = this.state;
			return (
        <div>
      <main>
        <Form onSubmit={this.onSubmit}>
					<Form.Field error={!!errors.email}>
							<label htmlFor="email">Email</label>
							<input
							 type="text" 
							 id="email" 
							 name="email" 
							 placeholder="email@email.com"
							 value={data.email}
							 onChange={this.onChange}
							 />
							{errors.email && <InlineError text={errors.email} />}
					</Form.Field>
					<Form.Field error={!!errors.email}> 
							<label htmlFor="password">password</label>
							<input
							 type="password" 
							 id="password" 
							 name="password" 
							 placeholder="Make it secure"
							 value={data.password}
							 onChange={this.onChange}
							 />
							{errors.password && <InlineError text={errors.password} />}
					</Form.Field>
					<input type="submit"  value="Login"/>
        </Form>
				<div class="signup">
		  		<h3>New to Maintenance Tracker? <Link to="/signup">Sign Up</Link></h3>
		  	</div>
      </main>
		</div>
			)
    }
    
}
UserForm.propTypes = {
	submit: propTypes.func.isRequired
};

 export default UserForm;