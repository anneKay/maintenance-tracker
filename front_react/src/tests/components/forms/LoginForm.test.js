import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import LoginForm from '../../../components/form/LoginForm';


describe(' Login Form component rendering', () => {
  const props = {
    submit: jest.fn(),
    history: {},
    error: 'unauthorized',
  };
  const emailEvent = {
    preventDefault() {},
    target: {
      name: 'email',
      value: 'example@email.com',
    },
  };
  const passwordEvent = {
    preventDefault() {},
    target: {
      name: 'password',
      value: 'yourpass',
    },
  };
  const sinonSpy = sinon.spy();
  const wrapper = shallow(<LoginForm onSubmit={sinonSpy} {...props} />);

  it('Should render properly', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call the submit method on submit', () => {
    wrapper.find('Form').simulate('submit', { preventDefault() {} });
    expect(sinonSpy.called).toBe(false);
  });

  it('Should call the onChange function and set email in state', () => {
    wrapper.instance().onChange(emailEvent);
    const actualEmail = wrapper.state().data.email;
    const expectedEmail = 'example@email.com';
    expect(actualEmail).toBe(expectedEmail);
  });
  it('Should call the onChange function and set password in the state', () => {
    wrapper.instance().onChange(passwordEvent);
    const actualPassword = wrapper.state().data.password;
    const expectedpassword = 'yourpass';
    expect(actualPassword).toBe(expectedpassword);
  });
  it('Should call the submit method on submit', () => {
    wrapper.find('Form').simulate('submit', { preventDefault() {} });
    expect(sinonSpy.called).toBe(false);
  });
  it('Should call the submit method on submit', () => {
    wrapper.find('Form').simulate('submit', { preventDefault() {} });
    wrapper.instance().onSubmit();
    expect(sinonSpy.called).toBe(false);
  });
  wrapper.instance().onSubmit();
});
