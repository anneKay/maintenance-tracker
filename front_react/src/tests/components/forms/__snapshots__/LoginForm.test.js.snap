// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Login Form component rendering Should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserForm
    error="unauthorized"
    history={Object {}}
    onSubmit={[Function]}
    submit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <main>
        <Message
          negative={true}
        >
          unauthorized
        </Message>
        <Form
          as="form"
          onSubmit={[Function]}
        >
          <FormField
            error={true}
          >
            <label
              htmlFor="email"
            >
              Email:
              <input
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Enter email address"
                required={true}
                type="text"
                value=""
              />
            </label>
            <Label
              color="red"
              pointing={true}
            >
              invalid email
            </Label>
          </FormField>
          <FormField
            error={false}
          >
            <label
              htmlFor="password"
            >
              Password:
              <input
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Enter your password"
                required={true}
                type="password"
                value=""
              />
            </label>
          </FormField>
          <input
            type="submit"
            value="Login"
          />
        </Form>
        <div
          className="signup"
        >
          <h3>
            New to Maintenance Tracker? 
            <Link
              replace={false}
              to="/signup"
            >
              Sign Up
            </Link>
          </h3>
        </div>
      </main>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Message
            negative={true}
          >
            unauthorized
          </Message>,
          <Form
            as="form"
            onSubmit={[Function]}
          >
            <FormField
              error={true}
            >
              <label
                htmlFor="email"
              >
                Email:
                <input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Enter email address"
                  required={true}
                  type="text"
                  value=""
                />
              </label>
              <Label
                color="red"
                pointing={true}
              >
                invalid email
              </Label>
            </FormField>
            <FormField
              error={false}
            >
              <label
                htmlFor="password"
              >
                Password:
                <input
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Enter your password"
                  required={true}
                  type="password"
                  value=""
                />
              </label>
            </FormField>
            <input
              type="submit"
              value="Login"
            />
          </Form>,
          <div
            className="signup"
          >
            <h3>
              New to Maintenance Tracker? 
              <Link
                replace={false}
                to="/signup"
              >
                Sign Up
              </Link>
            </h3>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "unauthorized",
            "negative": true,
          },
          "ref": null,
          "rendered": "unauthorized",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "form",
            "children": Array [
              <FormField
                error={true}
              >
                <label
                  htmlFor="email"
                >
                  Email:
                  <input
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter email address"
                    required={true}
                    type="text"
                    value=""
                  />
                </label>
                <Label
                  color="red"
                  pointing={true}
                >
                  invalid email
                </Label>
              </FormField>,
              <FormField
                error={false}
              >
                <label
                  htmlFor="password"
                >
                  Password:
                  <input
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter your password"
                    required={true}
                    type="password"
                    value=""
                  />
                </label>
              </FormField>,
              <input
                type="submit"
                value="Login"
              />,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="email"
                  >
                    Email:
                    <input
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter email address"
                      required={true}
                      type="text"
                      value=""
                    />
                  </label>,
                  <Label
                    color="red"
                    pointing={true}
                  >
                    invalid email
                  </Label>,
                ],
                "error": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Email:",
                      <input
                        id="email"
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter email address"
                        required={true}
                        type="text"
                        value=""
                      />,
                    ],
                    "htmlFor": "email",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Email:",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "defaultValue": undefined,
                        "id": "email",
                        "name": "email",
                        "onChange": [Function],
                        "placeholder": "Enter email address",
                        "required": true,
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "invalid email",
                    "color": "red",
                    "pointing": true,
                  },
                  "ref": null,
                  "rendered": "invalid email",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="password"
                  >
                    Password:
                    <input
                      id="password"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter your password"
                      required={true}
                      type="password"
                      value=""
                    />
                  </label>,
                  undefined,
                ],
                "error": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Password:",
                      <input
                        id="password"
                        name="password"
                        onChange={[Function]}
                        placeholder="Enter your password"
                        required={true}
                        type="password"
                        value=""
                      />,
                    ],
                    "htmlFor": "password",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Password:",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "defaultValue": undefined,
                        "id": "password",
                        "name": "password",
                        "onChange": [Function],
                        "placeholder": "Enter your password",
                        "required": true,
                        "type": "password",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "label",
                },
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Login",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              New to Maintenance Tracker? 
              <Link
                replace={false}
                to="/signup"
              >
                Sign Up
              </Link>
            </h3>,
            "className": "signup",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "New to Maintenance Tracker? ",
                <Link
                  replace={false}
                  to="/signup"
                >
                  Sign Up
                </Link>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "New to Maintenance Tracker? ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sign Up",
                  "replace": false,
                  "to": "/signup",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": [Function],
              },
            ],
            "type": "h3",
          },
          "type": "div",
        },
      ],
      "type": "main",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <main>
          <Message
            negative={true}
          >
            unauthorized
          </Message>
          <Form
            as="form"
            onSubmit={[Function]}
          >
            <FormField
              error={true}
            >
              <label
                htmlFor="email"
              >
                Email:
                <input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Enter email address"
                  required={true}
                  type="text"
                  value=""
                />
              </label>
              <Label
                color="red"
                pointing={true}
              >
                invalid email
              </Label>
            </FormField>
            <FormField
              error={false}
            >
              <label
                htmlFor="password"
              >
                Password:
                <input
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Enter your password"
                  required={true}
                  type="password"
                  value=""
                />
              </label>
            </FormField>
            <input
              type="submit"
              value="Login"
            />
          </Form>
          <div
            className="signup"
          >
            <h3>
              New to Maintenance Tracker? 
              <Link
                replace={false}
                to="/signup"
              >
                Sign Up
              </Link>
            </h3>
          </div>
        </main>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Message
              negative={true}
            >
              unauthorized
            </Message>,
            <Form
              as="form"
              onSubmit={[Function]}
            >
              <FormField
                error={true}
              >
                <label
                  htmlFor="email"
                >
                  Email:
                  <input
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter email address"
                    required={true}
                    type="text"
                    value=""
                  />
                </label>
                <Label
                  color="red"
                  pointing={true}
                >
                  invalid email
                </Label>
              </FormField>
              <FormField
                error={false}
              >
                <label
                  htmlFor="password"
                >
                  Password:
                  <input
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter your password"
                    required={true}
                    type="password"
                    value=""
                  />
                </label>
              </FormField>
              <input
                type="submit"
                value="Login"
              />
            </Form>,
            <div
              className="signup"
            >
              <h3>
                New to Maintenance Tracker? 
                <Link
                  replace={false}
                  to="/signup"
                >
                  Sign Up
                </Link>
              </h3>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "unauthorized",
              "negative": true,
            },
            "ref": null,
            "rendered": "unauthorized",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "form",
              "children": Array [
                <FormField
                  error={true}
                >
                  <label
                    htmlFor="email"
                  >
                    Email:
                    <input
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter email address"
                      required={true}
                      type="text"
                      value=""
                    />
                  </label>
                  <Label
                    color="red"
                    pointing={true}
                  >
                    invalid email
                  </Label>
                </FormField>,
                <FormField
                  error={false}
                >
                  <label
                    htmlFor="password"
                  >
                    Password:
                    <input
                      id="password"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter your password"
                      required={true}
                      type="password"
                      value=""
                    />
                  </label>
                </FormField>,
                <input
                  type="submit"
                  value="Login"
                />,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <label
                      htmlFor="email"
                    >
                      Email:
                      <input
                        id="email"
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter email address"
                        required={true}
                        type="text"
                        value=""
                      />
                    </label>,
                    <Label
                      color="red"
                      pointing={true}
                    >
                      invalid email
                    </Label>,
                  ],
                  "error": true,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Email:",
                        <input
                          id="email"
                          name="email"
                          onChange={[Function]}
                          placeholder="Enter email address"
                          required={true}
                          type="text"
                          value=""
                        />,
                      ],
                      "htmlFor": "email",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Email:",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "defaultValue": undefined,
                          "id": "email",
                          "name": "email",
                          "onChange": [Function],
                          "placeholder": "Enter email address",
                          "required": true,
                          "type": "text",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                    ],
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "invalid email",
                      "color": "red",
                      "pointing": true,
                    },
                    "ref": null,
                    "rendered": "invalid email",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <label
                      htmlFor="password"
                    >
                      Password:
                      <input
                        id="password"
                        name="password"
                        onChange={[Function]}
                        placeholder="Enter your password"
                        required={true}
                        type="password"
                        value=""
                      />
                    </label>,
                    undefined,
                  ],
                  "error": false,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Password:",
                        <input
                          id="password"
                          name="password"
                          onChange={[Function]}
                          placeholder="Enter your password"
                          required={true}
                          type="password"
                          value=""
                        />,
                      ],
                      "htmlFor": "password",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Password:",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "defaultValue": undefined,
                          "id": "password",
                          "name": "password",
                          "onChange": [Function],
                          "placeholder": "Enter your password",
                          "required": true,
                          "type": "password",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                    ],
                    "type": "label",
                  },
                  undefined,
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "type": "submit",
                  "value": "Login",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                New to Maintenance Tracker? 
                <Link
                  replace={false}
                  to="/signup"
                >
                  Sign Up
                </Link>
              </h3>,
              "className": "signup",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "New to Maintenance Tracker? ",
                  <Link
                    replace={false}
                    to="/signup"
                  >
                    Sign Up
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "New to Maintenance Tracker? ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign Up",
                    "replace": false,
                    "to": "/signup",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": [Function],
                },
              ],
              "type": "h3",
            },
            "type": "div",
          },
        ],
        "type": "main",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
